homeassistant/components/opentherm_gw/climate.py

```diff
@@ -138,16 +140,25 @@ def temperature_unit(self):
         """Return the unit of measurement used by the platform."""
         return TEMP_CELSIUS
 
+    @property
+    def hvac_action(self):
+        """Return current HVAC operation."""
+        return self._current_operation
+
     @property
     def hvac_mode(self):
         """Return current HVAC mode."""
-        return self._current_operation
+        return HVAC_MODE_AUTO
 
     @property
     def hvac_modes(self):
         """Return available HVAC modes."""
         return []
 
+    def set_hvac_mode(self, hvac_mode):
+        """Set the HVAC mode."""
+        _LOGGER.warning("Changing HVAC mode is not supported")
```

Comment:
pass that without warning, because you will see warnings on startup. `HVAC_MODE_AUTO` mean that you can't change temperature. Otherwise, you need to pin that as HVAC_MODE_HEAT or HVAC_MODE_HEAT_COOL

Reply:
I don't see any warnings on startup.

I'm open to suggestions regarding the `hvac_mode`. The problem is that those modes are purely a thermostat thing and they don't exist in the OpenTherm protocol, which is what the gateway is reading. There's no way to get this information from the actual thermostat.
Then there's the fact that the protocol supports both heating and cooling so `HVAC_MODE_HEAT` and `HVAC_MODE_COOL` by themselves aren't appropriate, but the target is a specific temperature rather than a range, so `HVAC_MODE_HEAT_COOL` is out as well.

As I see it, we're left with two options:
* Use `HVAC_MODE_HEAT` and `HVAC_MODE_COOL` depending on which mode is active (possibly with `HVAC_MODE_OFF` if neither is active)
* Use `HVAC_MODE_AUTO` at all times
I chose the latter as the first would just be a reflection of the `hvac_action` and as I understand it that's not the purpose of the `hvac_mode`

Reply:
Yes, hvac_mode is what is set on climate:
- Climate should cool only if needed: HVAC_MODE_COOL
- Climate should heat only if needed: HVAC_MODE_HEAT
- Climate should hold the temp in a confort zone: HVAC_MODE_HEAT_COOL
- Climate is control by a hub/AI/scheduler: HVAC_MODE_AUTO

Reply:
Thanks, that confirms my understanding of the various modes.
As I said before, the OpenTherm Gateway has no such concept. Those modes are based on actual room thermostat settings and they exist in the room thermostat only and influence how the room thermostat controls the boiler. The OpenTherm Gateway sits between the room thermostat and the boiler and thus can only read the control commands that are sent between them.
As a result, if the thermostat is asking the boiler to heat, we know that the `hvac_mode` on the room thermostat is likely to be `HVAC_MODE_HEAT` and if it's asking for cooling, it's likely to be `HVAC_MODE_COOL`. However, if it's not requesting either, we have no way of knowing the `hvac_mode` of the room thermostat.
I would still say that `HVAC_MODE_AUTO` is closest to reality. The climate is controlled by a hub/AI/scheduler (the actual room thermostat), the only difference is that we have the option to override the temperature setting from HA. The climate component itself does not prevent us from changing the temperature while `HVAC_MODE_AUTO` is active.

Reply:
If you can active cool rooms (cool water in boiler), you need to use `HVAC_MODE_HEAT_COOL`. If you can only heating, you need to use `HVAC_MODE_HEAT`. HVAC_MODE_AUTO is wrong. Thermostats are only not an AI.

![image](https://user-images.githubusercontent.com/15338540/63092119-e6b68e00-bf60-11e9-81e1-438e895ba2c0.png)

Do you live in Europe? In that case, most climate is only HVAC_MODE_HEAT.

Reply:
My own thermostat is indeed `HVAC_MODE_HEAT` only, but the OpenTherm protocol supports cooling signals as well and I'm trying to make it work for any situation.
I can change the implementation to report `HVAC_MODE_HEAT` when heating is active and `HVAC_MODE_COOL` when cooling is active - basically reflecting the `hvac_action` - but in this case I am unsure what to do when the system is idle. I see 3 options:
* Default to `HVAC_MODE_HEAT` (or `HVAC_MODE_COOL`)
* Keep the mode related to the last `hvac_action`
* Report `HVAC_MODE_OFF`

Neither is perfect and in all cases the thermostat will deviate in some situations. Which would you say is most appropriate?

Reply:
Updated to publish the `hvac_mode` related to the last active `hvac_action`, assuming `HVAC_MODE_HEAT` on start.